frontend/src/Connection.ts

```diff
@@ -565,3 +588,26 @@ export class Connection {
     this.bindSocket();
   };
 }
+
+/**
+ * Fire a function if the timer runs out. You can reset it, or
+ * cancel it to prevent the function from being fired.
+ *
+ * @param onExpired
+ * @param timeoutMs
+ * @returns
+ */
+function resettableTimeout(onExpired: () => void, timeoutMs: number) {
+  let timer = setTimeout(onExpired, timeoutMs);
+
+  return {
+    reset: () => {
+      clearTimeout(timer);
+      timer = setTimeout(onExpired, timeoutMs);
+    },
+    cancel: () => {
+      clearTimeout(timer);
+    },
```

Comment:
No `this` so can just do regular method shorthand.

```suggestion
    reset() {
      clearTimeout(timer);
      timer = setTimeout(onExpired, timeoutMs);
    },
    cancel() {
      clearTimeout(timer);
    },
```

Reply:
Ah true, I always forget about this shorthand!

(I also forget a bunch of the syntax things around classes, since I do my best to avoid them at all costs :D